source/develop/release-management/features/virt/dedicated-cpu.md

```diff
@@ -0,0 +1,170 @@
+* Email: tgolembi@redhat.com
+
+
+## Description
+
+To be able to have more flexibility in configure how virtual CPUs (vCPUs)
+of VMs use physical CPUs (pCPUs) of the host this feature introduces several
+policies for dedicating CPUs. Compared to CPU pinning the advantages are that:
+
+  * VM does not need to be pinned to a specific host,
+  * solves some of the issues with *existing* pinning policy where
+    several VMs can end up to be pinned to same CPUs,
+  * *TODO* I am pretty sure there were couple more worth mentioning here
+
+The available policies described below divide the set of available pCPUs into
+three sub-sets, There are the dedicated CPUs that are assigned only to a single
+VM. Then there are some unassigned CPUs that cannot be used by any other VM
+because of restrictions requirements on the threads of the core. And finally
+there is the set of CPUs that are not strictly assigned to any particular VM
+and are shared between all the VMs with *shared* policy. This subset is hereby
+referred to as *shared pool*.
+
+The available CPU policies are:
+
+  * *shared*: *(default)* The guest vCPUs will be allowed to roam among all
+    pCPUs from shared pool.
+  * *dedicated*: The guest vCPUs will be strictly pinned to a set of host pCPUs
+    (similarly to static CPU pinning). The set of pCPUs will be chosen to match
+    the required guest CPU topology. If the host has SMT architecture, thread
+    siblings are preferred.
+  * *isolate-threads*: If the host does not have SMT architecture this is same
+    as *dedicated* policy. If the host has SMT architecture each vCPU is placed
+    on a different physical core and all thread siblings are excluded from use
+    by this and any other VM (rendering all but one thread from a core unusable).
+  * *require-threads*: The host must have an SMT architecture. Each vCPU is
+    allocated on thread siblings and threads from single core can only be
+    assigned to this VM. If some threads remain unused these will be excluded
+    from use by any other VM.
+
+
+## Prerequisites
+
+There are no special prerequisites for this feature.
+
+## Security
+
+This feature is primarily designed for performance tuning and is not meant to
+be a security feature. It is possible for other VMs to share the dedicated CPUs
+for a short period of time when the VM is brought up (boot, after migration,
+restore from snapshots, etc.). This is not so much problem during boot because
+by the time OS is started the CPUs should be already evicted, but it could
+potentially provide a window of opportunity (for exploiting CPU
+vulnerabilities) after finished migration or when restoring VM from snapshot.
+
+## Limitations
+
+See the *Security* section.
+
+Currently, when some CPUs are already dedicated to a VM they cannot be evicted
+and provided to another VM. Notably this means:
+
+  * it may be impossible to start a VMs with CPU pinning on that host because
+    the requested CPUs are already occupied;
```

Comment:
right, and it should be prevented on the engine side, right?
and "a VM" ?

Reply:
yes